src/Xamarin.Android.Build.Tasks/Tasks/ResolveAssemblies.cs

```diff
@@ -201,17 +206,42 @@ void AddAssemblyReferences (DirectoryAssemblyResolver resolver, ICollection<stri
 			}
 
 			LogMessage ("{0}Adding assembly reference for {1}, recursively...", new string (' ', indent), assembly.Name);
+			resolutionPath.Add (assembly.Name.Name);
 			indent += 2;
+
 			// Add this assembly
 			if (!topLevel && assemblies.All (a => new AssemblyNameDefinition (a, null).Name != assembly.Name.Name))
 				assemblies.Add (fullPath);
 
 			// Recurse into each referenced assembly
 			foreach (AssemblyNameReference reference in assembly.MainModule.AssemblyReferences) {
-				var reference_assembly = resolver.Resolve (reference);
-				AddAssemblyReferences (resolver, assemblies, reference_assembly, false);
+				AssemblyDefinition reference_assembly;
+				try {
+					reference_assembly = resolver.Resolve (reference);
+				} catch (FileNotFoundException ex) {
+					var references = new StringBuilder ();
+					for (int i = resolutionPath.Count - 1; i >= 0; i--) {
+						references.Append ('`');
+						references.Append (resolutionPath [i]);
+						references.Append ('`');
+						if (i != 0)
+							references.Append (" > ");
```

Comment:
Please add a test for the multiple references scenario, if possible. My (bad?) reading of the `for` loop makes me wonder if this will be *backwards* from what I wanted, going from "missing" to "top-level", instead of from top-level to missing.